Fix extension for 4.0

raulsntos opened the pull request:
- The `.mono` folder was moved to `.godot/mono` (https://github.com/godotengine/godot/pull/38704)
- The `--remote-debug` argument now requires the protocol (https://github.com/godotengine/godot/pull/37099)
- The `-q` argument has been repurposed, now `--quit` should be used (https://github.com/godotengine/godot/pull/55696)

Although these changes should fix the extension for 4.0 while keeping support for 3.x, they feel hacky and it might be better to simply provide a configuration setting for the user that defaults to the current stable version of Godot instead of trying to detect the version.

Closes #38

neikeq commented:
In a week, we'll be merging the `dotnet6` branch to `master` and this will no longer work. I can merge this if still desired, though.

raulsntos commented:
I don't see how these specific changes are related to that merge, unless you mean that Godot 4.0 will be unusable due to the many things that were commented out and that need to be rewritten, in which case I still think these changes would eventually be needed, no?

Either way, I'm in no hurry to merge this as 4.0 is still in alpha, just looked into it because somebody asked and thought I'd leave this here for when we need it.

neikeq commented:
> I don't see how these specific changes are related to that merge

The extension uses the Mono debugger, which won't be usable in 4.x because the editor (including the player) uses CoreCLR (as it uses the runtime from the .NET Sdk). This means the VS debugger won't be able to connect, which I think results in a launch error.

raulsntos committed:
Fix extension for 4.0

- The `.mono` folder was moved to `.godot/mono`
- The `--remote-debug` argument now requires the protocol
- The `-q` argument has been repurposed, now `--quit` should be used

Kezzo commented:
Hey! ✌️
What's the status of this PR?

Godot 4.0 seems to be getting closer to release and is in bugfixing mode now. So there shouldn't be any big changes on the C# side anymore.
If we don't want to deal with backwards compatibility for Godot 3.x right now, we could still release a preview package of this with Godot 4.0 support to get feedback and bug reports already coming in!

raulsntos commented:
Superseded by https://github.com/godotengine/godot-csharp-vscode/pull/56
